includes/class-amp-reader-themes.php

```diff
@@ -0,0 +1,546 @@
+				return in_array( $key, $keys, true );
+			},
+			ARRAY_FILTER_USE_KEY
+		);
+
+		foreach ( $keys as $key ) {
+			if ( ! array_key_exists( $key, $prepared_theme ) ) {
+				$prepared_theme[ $key ] = '';
+			}
+		}
+
+		return $prepared_theme;
+	}
+
+	/**
+	 * Provides the current theme name.
+	 *
+	 * @return string|bool The theme name, or false if the theme has errors.
+	 */
+	private function get_current_theme_name() {
+		if ( is_null( $this->current_theme_name ) ) {
+			$current_theme = wp_get_theme();
+
+			$this->current_theme_name = $current_theme->exists() ? $current_theme->get( 'Name' ) : false;
+		}
+
+		return $this->current_theme_name;
+	}
+
+	/**
+	 * Returns whether a theme can be installed on the system.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param array $theme Theme data.
+	 * @return bool True if themes can be installed.
+	 */
+	public function can_install_theme( $theme ) {
+		// @todo Add support for installing non-default reader themes. Until that is done, themes that are provided via
+		// the amp_reader_themes filter will show on the reader themes screen but will need to be manually installed on
+		// the site.
+		$default_reader_theme_slugs = wp_list_pluck( $this->get_default_raw_reader_themes(), 'slug' );
+		if ( ! in_array( $theme['slug'], $default_reader_theme_slugs, true ) ) {
+			return false;
+		}
+
+		if ( is_null( $this->can_install_themes ) ) {
+			if ( ! class_exists( 'WP_Upgrader' ) ) {
+				require_once ABSPATH . 'wp-admin/includes/file.php';
+				require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+			}
+
+			$this->can_install_themes = true === ( new WP_Upgrader() )->fs_connect( [ get_theme_root() ] );
+		}
+
+		if ( ! $this->can_install_themes ) {
+			return false;
+		}
+
+		if ( ! empty( $theme['requires'] ) && ! version_compare( get_bloginfo( 'version' ), $theme['requires'], '>=' ) ) {
+			return false;
+		}
+
+		if ( ! empty( $theme['requires_php'] ) && ! version_compare( phpversion(), $theme['requires_php'], '>=' ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Returns reader theme availability status.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param array $theme Theme data.
+	 * @return string Theme availability status.
+	 */
+	public function get_theme_availability( $theme ) {
+		switch ( true ) {
+			case $this->get_current_theme_name() === $theme['name']:
+				return self::ACTIVE_STATUS;
+
+			case wp_get_theme( $theme['slug'] )->exists():
+				return self::INSTALLED_STATUS;
+
+			case $this->can_install_theme( $theme ):
+				return self::INSTALLABLE_STATUS;
+
+			default:
+				return self::NON_INSTALLABLE_STATUS;
+		}
+	}
+
+	/**
+	 * Provides details for the classic theme included with the plugin.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @return array
+	 */
+	private function get_classic_mode() {
+		return [
+			'name'           => 'AMP Classic',
+			'slug'           => 'classic',
+			'preview_url'    => 'https://amp-wp.org',
+			'screenshot_url' => '//via.placeholder.com/218x472',
```

Comment:
This is for testing purposes I suppose.

Reply:
Yeah, it's filtered in the REST controller, but I'm going to make it an empty string here.